### PR TITLE
Added basic obstacle avoidance.

### LINKED ISSUE #1: Obstacle avoidance
As a driver i want to be notified when the car notices obstacles in proximity.

Acceptence critera:
- The car should should stop ahead of an obstacle.

### PR DESCRIPTION
**Pull request by linusaaberg**
Added basic obstacle avoidance to fix issue #1 .

### CODE REVIEW

File: SmartCar/src/Basic_Obstacle_Avoidance.ino [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+const int ECHO_PIN=18;
+const unsigned int MAX_DISTANCE = 100;
+SR04 front(TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+
+void setup(){
+  Serial.begin(9600);
+  }
+
+void loop() {
+  while(front.getDistance()==0 || front.getDistance() > 15){
```
Comment (platisd) [inline SmartCar/src/Basic_Obstacle_Avoidance.ino]: This conducts **two** measurements that may be different. If for example the first measurement is 100 and the next one is 0 (which may very well happen if there aren't any obstacles nearby) then your car will stop.
Additionally a 0 may be returned if there's something too close.
Finally, why have two loops inside each other? loop() already... Loops!

In other words there are 3 issues at line 19.:)

**Review by platisd — CHANGES_REQUESTED** [review]
Fix the problems and consider having the issue that is fixed in the pull request's description, not title.

**Comment by linusaaberg** [comment]
Fixes issue #1 . Had to check for guidance on the internet. Found this site and tinkered with the code a bit: https://create.arduino.cc/projecthub/Isaac100/getting-started-with-the-hc-sr04-ultrasonic-sensor-036380. Also added a drive function to runt in the loop.

**Review by platisd — CHANGES_REQUESTED** [review]
Still not there. To begin with, you should use the Smartcar library to control the sensor. Secondly, the logic is still wrong. If there is no obstacle, your car will stop moving.

**Review by platisd — APPROVED** [review]

**Pull request merged**